### PR TITLE
NEXT-37040: Subscription documentation

### PR DESCRIPTION
**Pull request by Bird87ZA**

### CODE REVIEW

File: products/extensions/subscriptions/guides/request-scoping.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+           controller="subscription.storefront.controller.checkout::cartPage">
+        <default key="_noStore">true</default>
+        <default key="_routeScope"><list><string>storefront</string></list></default>
+        <default key="_subscriptionCart">true</default>
+        <default key="_subscriptionContext">true</default>
+        <default key="_controllerName">checkout</default>
+        <default key="_controllerAction">cartpage</default>
+        <default key="_templateScopes">subscription</default>
+        <option key="seo">false</option>
+        <condition>service('license').check('SUBSCRIPTIONS-2437281')</condition>
```
Comment (mstegmeyer) [inline products/extensions/subscriptions/guides/request-scoping.md:27]: maybe remove, not necessary for extensions

File: products/extensions/subscriptions/guides/index.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+nav:
+  title: Guides
+  position: 20
+
+---
+
+# Guides
+
+This section will help with common topics regarding usage of subscriptions.
```
[the remark this answers is not in the document]
Comment (Bird87ZA) [reply products/extensions/subscriptions/guides/index.md:10]: I've had a look at the other guides, and they are also one liners. I think the links added to these pages are generated by whatever is presenting it.

File: products/extensions/subscriptions/concept.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Concepts
+
+To use subscriptions, you will need to be familiar with two core concepts in subscription models.
+
+## Plans
+
+A plan is a set of rules that define the subscription. This includes the price, the billing interval, and the product that the customer will receive. Plans can be created and managed in the Shopware administration.
```
Comment (lernhart) [inline products/extensions/subscriptions/concept.md:7]: A plan may have multiple intervals assigned

**Review by lernhart — CHANGES_REQUESTED** [review]

**Comment by lernhart** [comment]
@mstegmeyer do you think we should document the interval logic (relative, absolute, cron stuff, etc.) ?

**Comment by Bird87ZA** [comment]
> @mstegmeyer do you think we should document the interval logic (relative, absolute, cron stuff, etc.) ?

I think it's a good idea. I shall do it.

File: products/extensions/subscriptions/concept.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+Intervals can be of three different types:
+
+### Relative
+
+A relative interval is an interval that is determined by a previous interval. For example, if a customer subscribes to a monthly plan, the next interval will be one month after the first delivery. These intervals are determined by using PHP's `DateInterval` class.
+
+### Absolute
+
+An absolute interval is an interval that is determined by a fixed date. For example, if a customer subscribes to a monthly plan, the next interval will be on the same day of the month as the first delivery. These work with cron expressions.
```
Comment (mstegmeyer) [inline products/extensions/subscriptions/concept.md:21]: No, that's relative again, it's **_relative_** to the start date of the subscription. Absolut would be "the first of the month" or "Every Wednesday"

File: products/extensions/subscriptions/guides/request-scoping.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+---
+nav:
+  title: Request scoping
+  position: 20
+
+---
+
+# Request scoping
+
+When you are in a subscription checkout, you are using a separate cart and context. In Storefront, there is an additional URL parameter (`subscriptionToken`) that gets resolved. In headless, there are four header parameters that need to be set namely `sw-subscription-plan`, `sw-subscription-interval`, `_subscriptionCart` and `_subscriptionContext`.
```
Comment (mstegmeyer) [inline products/extensions/subscriptions/guides/request-scoping.md:10]: The parameters with a `_` are not in the header, but are `attributes` of the request. They tell the resolvers to replace the object with the subscription ones.

File: products/extensions/subscriptions/guides/request-scoping.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+And, here is an example of the headers set on a subscription cart using headless:
+
+```bash
+curl -XPOST /
+    -H 'sw-subscription-plan: planId' /
+    -H 'sw-subscription-interval: intervalId' /
+    -H '_subscriptionCart: true' /
+    -H '_subscriptionContext: true' /
```
Comment (mstegmeyer) [inline products/extensions/subscriptions/guides/request-scoping.md:37]: don't think this is necessary

File: products/extensions/subscriptions/guides/template-scoping.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+In a subscription context, it's important to ensure that certain template adjustments, which are applicable to the standard storefront, are not automatically applied. This precaution helps in maintaining a clear distinction between the regular checkout process and the subscription checkout process. For instance, elements or buttons that facilitate immediate purchases or third-party payment options, like PayPal Express, should not be visible during the subscription checkout to avoid confusion.
+
+To achieve this separation, templates used within the subscription context should explicitly define their scope. Below is an example of extending a template in the default and subscription context:
+
+```twig
+{% sw_extends {
+    template: '@Storefront/storefront/base.html.twig',
+    scopes: ['default', 'subscription']
+} %}
```
Comment (mstegmeyer) [inline products/extensions/subscriptions/guides/template-scoping.md:18]: I think, a closing ``` is missing here ;)

**Review by mstegmeyer — APPROVED** [review]

**Comment by Isengo1989** [comment]
Thanks for your PR @Bird87ZA :tada:

**Pull request merged**